### PR TITLE
Refactoring CMakeLists for unit tests in folder application_manager

### PR DESCRIPTION
**Pull request by BSolonenko**
This PR is **[ready]** for review.

### Risk
This PR makes **[no]** API changes.

### Summary
These changes include next tests into a build: 
- vi_commands_test
- sdl_commands_test
- rc_commands_test
- rc_plugin_test
- application_manager_test
- application_manager_mock_hmi_test
- help_prompt_manager_test
- request_controller_test
- data_resumption_test
- state_controller_test
- app_launch_ctrl_test
- app_launch_data_test
- commands_test
- message_helper_test
- config_profile_test
- connection_handler_test
- formatters_test

##### Breaking Changes

1. Included next folders into the build:
- application_manager/rpc_plugins/rc_rpc_plugin/test
- application_manager/rpc_plugins/sdl_rpc_plugin/test
- application_manager/rpc_plugins/vehicle_info_plugin/test
- application_manager/test
- application_manager/test/app_launch
- application_manager/test/commands
- application_manager/test/message_helper
- application_manager/test/state_controller
2. Changed appropriate CMakeLists.txt 
3. Fixed path to headers in some sources
4. Added helpers for creating static and dynamic libraries in plugins

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/helpers.cmake [hunk 1]
```diff
@@ -0,0 +1,50 @@
+function(add_dynamic_library NAME)
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply src/components/application_manager/rpc_plugins/helpers.cmake]: @AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/392268c4b62313fe0e8988de899ca6c0317ab51b

File: src/components/application_manager/test/app_launch/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# Copyright (c) 2016, Ford Motor Company All rights reserved.
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply src/components/application_manager/test/app_launch/CMakeLists.txt]: @AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/392268c4b62313fe0e8988de899ca6c0317ab51b

File: src/components/application_manager/test/commands/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2016, Ford Motor Company All rights reserved.
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply src/components/application_manager/test/commands/CMakeLists.txt]: @AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/392268c4b62313fe0e8988de899ca6c0317ab51b

File: src/components/application_manager/test/message_helper/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2015, Ford Motor Company All rights reserved.
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply src/components/application_manager/test/message_helper/CMakeLists.txt]: @AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/392268c4b62313fe0e8988de899ca6c0317ab51b

File: src/components/application_manager/test/state_controller/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,44 @@
+# Copyright (c) 2015, Ford Motor Company All rights reserved.
```
[the remark this answers is not in the document]
Comment (BSolonenko) [reply src/components/application_manager/test/state_controller/CMakeLists.txt]: @AKalinich-Luxoft Fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/392268c4b62313fe0e8988de899ca6c0317ab51b

**Comment by BSolonenko** [comment]
@AKalinich-Luxoft Please, review again.

File: src/components/application_manager/rpc_plugins/CMakeLists.txt [hunk 6]
```diff
@@ -26,6 +26,7 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 # --- Utils
+include("./helpers.cmake")
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/CMakeLists.txt]: @BSolonenko can you specify just `include("helpers.cmake")` ?

File: src/components/application_manager/rpc_plugins/helpers.cmake [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+  target_link_libraries(${NAME}
+                        PUBLIC rpc_base::rpc_base
+                               utils::utils
+                               smart_objects::smart_objects
+                               interfaces::HMI_API
+                               interfaces::MOBILE_API
+                               interfaces::v4_protocol_v1_2_no_extra
+                               connection_handler::connection_handler
+                        PRIVATE application_manager::interface
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/helpers.cmake]: @BSolonenko why this are private?

File: src/components/application_manager/rpc_plugins/helpers.cmake [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+function(add_dynamic_library NAME)
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/helpers.cmake]: @BSolonenko please rename to add rpc_plugin. dynamic_library is to common. Content of this function is to specific

File: src/components/application_manager/rpc_plugins/helpers.cmake [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+          OWNER_READ
+          OWNER_WRITE
+          OWNER_EXECUTE
+          GROUP_READ
+          GROUP_EXECUTE
+          WORLD_READ
+          WORLD_EXECUTE)
+endfunction()
+
+function(add_static_library NAME)
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/helpers.cmake]: @BSolonenko for what do we use static lib? Is it also plugin? Also please make function name the same specific as function body.

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/CMakeLists.txt [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+  "${CMAKE_CURRENT_LIST_DIR}/commands/on_interior_vehicle_data_notification_test.cc"
+  "${CMAKE_CURRENT_LIST_DIR}/commands/on_remote_control_settings_test.cc"
+  "${CMAKE_CURRENT_LIST_DIR}/commands/rc_get_interior_vehicle_data_consent_test.cc"
+  "${CMAKE_CURRENT_LIST_DIR}/commands/set_interior_vehicle_data_request_test.cc"
+  )
+
+target_link_libraries(${PROJECT_NAME}
+                      PRIVATE components::interface rpc_plugins_static_lib)
+
+project(rc_plugin_test)
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/CMakeLists.txt]: @BSolonenko 2 projects in one file? Confuding. Avoid it please.

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+project(sdl_commands_test)
+
+create_test(${PROJECT_NAME})
+
+file(GLOB SOURCES
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/CMakeLists.txt]: @BSolonenko please avoid globing sources if is is possible

File: src/components/CMakeLists.txt [hunk 12]
```diff
@@ -48,14 +51,15 @@ else()
   target_include_directories(
     ${PROJECT_NAME}
     INTERFACE "${COMPONENTS_DIR}/include/policy/policy_regular")
+  set(POLICY_PATH ${COMPONENTS_DIR}/policy/policy_regular/)
```
Comment (AByzhynar) [inline src/components/CMakeLists.txt]: @BSolonenko What if policy flow set - HTTP?

File: src/components/application_manager/rpc_plugins/vehicle_info_plugin/test/CMakeLists.txt [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+project(vi_commands_test)
+
+create_test(${PROJECT_NAME})
+
+file(GLOB SOURCES
+          ${COMPONENTS_DIR}/application_manager/test/mock_message_helper.cc
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/vehicle_info_plugin/test/CMakeLists.txt]: @BSolonenko Strange alignment

**Comment by AByzhynar** [comment]
@BSolonenko All CMakeList.txt files are hardly readable due to different indentation in every file. Please make similar indentations across all files. Use ` src/components/application_manager/test/app_launch/CMakeLists.txt` as example.

File: src/components/CMakeLists.txt (hunk 12 above)
Comment (LuxoftAKutsan) [reply src/components/CMakeLists.txt]: @BSolonenko @AByzhynar  HTTP policy mode is in policy regular path

File: src/components/application_manager/rpc_plugins/CMakeLists.txt (hunk 6 above)
Comment (BSolonenko) [reply src/components/application_manager/rpc_plugins/CMakeLists.txt]: @LuxoftAKutsan fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/d4ee98f8065da4e11821acbf9f6d8a36940424bd

File: src/components/application_manager/rpc_plugins/helpers.cmake (hunk 7 above)
Comment (BSolonenko) [reply src/components/application_manager/rpc_plugins/helpers.cmake]: @LuxoftAKutsan fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/d4ee98f8065da4e11821acbf9f6d8a36940424bd

File: src/components/application_manager/rpc_plugins/helpers.cmake (hunk 8 above)
Comment (BSolonenko) [reply src/components/application_manager/rpc_plugins/helpers.cmake]: @LuxoftAKutsan fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/d4ee98f8065da4e11821acbf9f6d8a36940424bd

File: src/components/application_manager/rpc_plugins/helpers.cmake (hunk 9 above)
Comment (BSolonenko) [reply src/components/application_manager/rpc_plugins/helpers.cmake]: @LuxoftAKutsan fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/d4ee98f8065da4e11821acbf9f6d8a36940424bd

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/CMakeLists.txt (hunk 10 above)
Comment (BSolonenko) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/CMakeLists.txt]: @LuxoftAKutsan fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/d4ee98f8065da4e11821acbf9f6d8a36940424bd

File: src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/CMakeLists.txt (hunk 11 above)
Comment (BSolonenko) [reply src/components/application_manager/rpc_plugins/sdl_rpc_plugin/test/CMakeLists.txt]: @LuxoftAKutsan fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/d4ee98f8065da4e11821acbf9f6d8a36940424bd

**Comment by BSolonenko** [comment]
@LuxoftAKutsan All you comments fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/d4ee98f8065da4e11821acbf9f6d8a36940424bd. Please review again.

File: src/components/application_manager/test/commands/CMakeLists.txt [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+               ${CMAKE_CURRENT_SOURCE_DIR}/command_request_impl_test.cc
+               ${CMAKE_CURRENT_SOURCE_DIR}/command_response_impl_test.cc)
+
+target_include_directories(${PROJECT_NAME}
+                           PRIVATE ${APPLICATION_MANAGER_DIR}/test/include/)
+
+target_link_libraries(${PROJECT_NAME}
+                      PRIVATE application_manager::application_manager
+                              components::interface)
+
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/CMakeLists.txt]: @BSolonenko 
Please add an empty line here

Comment (BSolonenko) [reply src/components/application_manager/test/commands/CMakeLists.txt]: @AByzhynar fixed in https://github.com/smartdevicelink/sdl_core/pull/2780/commits/a9f525a5868a5e65e48a3a146865318cd6976e16

**Pull request closed without merging**